docs/shared/recipes/running-tasks/customizing-outputs.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Cache Results with Outputs
+description: 'Learn how to optimize what Nx restores from the cache'
+---
+
+# Cache Results with Outputs
+
+Whenever Nx runs a task which can be cached, it will store the results of that task in the cache.
+If nothing has changed, when Nx runs the task again, it will restore the results from the cache instead of spending the time to run the task again.
+
+## Types of Outputs
+
+### Terminal Output
+
+The terminal output of a task is replayed whenever a task is pulled from cache. Nx will always cache the terminal output of tasks which are cached.
+
+### Output Files
+
+Targets can define which tasks produce as a result. Nx will cache these files so that it can restore them when the task is pulled from cache.
```

Comment:
```suggestion
Targets can define which files are produced when a task is run. Nx will cache these files so that it can restore them when the task is pulled from cache.
```